simulator.go

```diff
@@ -266,14 +266,26 @@ func (h *simulatorAPIHandler) ServeHTTP(w http.ResponseWriter, r *http.Request)
 				http.Error(w, err.Error(), http.StatusInternalServerError)
 				return
 			}
-			logger = logger.New("id", container.ID[:8])
+			cLogger := logger.New("id", container.ID[:8])
 
 			logfile := fmt.Sprintf("client-%s.log", container.ID[:8])
-			if _, err = runContainer(h.daemon, container.ID, logger, filepath.Join(h.logdir, logfile), false); err != nil {
+			waiter, err := runContainer(h.daemon, container.ID, cLogger, filepath.Join(h.logdir, logfile), false)
+			if err != nil {
 				logger.Error("failed to start client", "error", err)
 				http.Error(w, err.Error(), http.StatusInternalServerError)
 				return
 			}
+			go func() {
```

Comment:
That means we'll end up with a ton of idle goroutines (up to two for every node we start, depending on the loglevel used) lingering around until the simulator node is terminated, and since we'd only close them at the end, the log files would remain open until then and we'd probably still hit the open file limit and crash, which is what I'm trying to fix.

> You'd also need to add the wait logic to the node DELETE handler https://github.com/karalabe/hive/blob/master/simulator.go#L340.

I don't understand why this would be necessary.